src/qibolab/transpilers/general_connectivity.py

```diff
@@ -380,13 +450,80 @@ def add_swaps(self, path, meeting_point):
         backward = path[meeting_point + 1 :: -1]
         if len(forward) > 1:
             for f1, f2 in pairwise(forward):
-                self.transpiled_circuit.add(gates.SWAP(self._qubit_map[f1], self._qubit_map[f2]))
+                self._transpiled_circuit.add(gates.SWAP(self._qubit_map[f1], self._qubit_map[f2]))
         if len(backward) > 1:
             for b1, b2 in pairwise(backward):
-                self.transpiled_circuit.add(gates.SWAP(self._qubit_map[b1], self._qubit_map[b2]))
+                self._transpiled_circuit.add(gates.SWAP(self._qubit_map[b1], self._qubit_map[b2]))
 
     def update_qubit_map(self):
         """Update the qubit mapping after adding swaps"""
         old_mapping = self._qubit_map.copy()
         for key, value in self._mapping.items():
             self._qubit_map[value] = old_mapping[key]
+
+
+def translate_circuit(circuit, two_qubit_natives, translate_single_qubit=False):
+    """Translates a circuit to native gates.
+
+    Args:
+        circuit (qibo.models.Circuit): Circuit model to translate into native gates.
+        two_qubit_natives (list): List of two qubit native gates
+            supported by the quantum hardware ("CZ" and/or "iSWAP").
+        translate_single_qubit (bool):
+
+    Returns:
+        new (qibo.models.Circuit): Equivalent circuit with native gates.
+    """
+    new = circuit.__class__(circuit.nqubits)
+    for gate in circuit.queue:
+        if len(gate.qubits) > 1 or translate_single_qubit:
+            new.add(translate_gate(gate, two_qubit_natives))
+        else:
+            new.add(gate)
+    return new
+
+
+# TODO
+def can_execute(circuit, two_qubit_natives, connectivity, verbose=True):
+    """Checks if a circuit can be executed on Hardware.
+
+    Args:
+        circuit (qibo.models.Circuit): Circuit model to check.
+        two_qubit_natives (list): List of two qubit native gates
+            supported by the quantum hardware ("CZ" and/or "iSWAP").
+        connectivity:
+        verbose (bool): If ``True`` it prints debugging log messages.
+
+    Returns ``True`` if the following conditions are satisfied:
+        - Circuit does not contain more than two-qubit gates.
+        - All one-qubit gates are I, Z, RZ or U3.
+        - All two-qubit gates are CZ or iSWAP based on two_qubit_natives.
+        - Circuit matches connectivity.
+    otherwise returns ``False``.
+    """
+
+    # pring messages only if ``verbose == True``
+    vlog = lambda msg: log.info(msg) if verbose else lambda msg: None
+    for gate in circuit.queue:
+        if isinstance(gate, gates.M):
+            continue
+
+        if len(gate.qubits) == 1:
+            if not isinstance(gate, (gates.I, gates.Z, gates.RZ, gates.U3)):
+                vlog(f"{gate.name} is not a single qubit native gate.")
+                return False
+
+        elif len(gate.qubits) == 2:
+            if gate.__class__.__name__ not in two_qubit_natives:
```

Comment:
```suggestion
            if type(gate).__name__ not in two_qubit_natives:
```

But also:
```suggestion
            if type(gate) not in two_qubit_natives:
```
and list classes, instead of their names (they are identifiers, known to language analyzers).